lasagne/layers/merge.py

```diff
@@ -91,7 +128,7 @@ def get_output_for(self, inputs, **kwargs):
             if coeff != 1:
                 input *= coeff
             if output is not None:
-                output += input
+                output = self.merge_function(output, input)
             else:
                 output = input
         return output
```

Comment:
If you feel strongly about its fine :) There is no software engineer inside me.

I see your point with coupling.